src/transformers/cache_utils.py

```diff
@@ -1652,9 +1652,7 @@ class HybridCache(Cache):
         ```
     """
 
-    # TODO (joao): dive deeper into gemma2 and paligemma -- there are reports of speed loss with compilation. Revert
-    # ALL changes from the PR that commented the line below when reactivating it.
-    # is_compileable = True
+    is_compileable = True
```

Comment:
nice! Can we update the cache to also init the layers lazily like we dofor HybridChunked cache?

Reply:
@ArthurZucker `HybridChunkedCache` only works if we don't compile the first forward pass, `HybricCache` works regardless of we compile the first forward pass or not. `torch._dynamo.mark_static_address` can't be called inside `torch.compile`, which lazy init does.

This means that if a user creates their own custom code with `HybridChunkedCache`, they can't simply compile the forward pass. If anything, `HybridChunkedCache` should move away from lazy init :P

Reply:
Chatted offline: 
1. lazy init is needed for TP
2. however, lazy init is incompatible with compiling the first forward pass (prefill). lazy init + `@torch.compiler.disable() `doesn't solve it either
3. solution: add a new flag `lazy_init = None`. If `torch.distributed` is initialized and the flag is unset, then it will be `True`.
4. Apply this change to ALL caches -> ALL caches compatible with TP